Add changelog for 2.0.0-alpha.14 (#562)
clusterio/clusterio: CHANGELOG.md

- Added `clusterio_host_system_info` metric with kernel, instruction set, CPU Model, hostname and Node.js version of hosts.
- Added `system_cpu_seconds_total`, `system_cpu_idle_seconds_total`, `system_memory_capasity_bytes`, `system_memory_available_bytes`, `system_disk_capasity_bytes`, `system_disk_available_bytes` metrics for CPU, memory and disk usage on the controller.
- Added `clusterio_host_system_cpu_seconds_total`, `clusterio_host_system_cpu_idle_seconds_total`, `clusterio_host_system_memory_capasity_bytes`, `clusterio_host_system_memory_available_bytes`, `clusterio_host_system_disk_capasity_bytes`, `clusterio_host_system_disk_available_bytes` metrics for CPU, memory and disk usage on hosts.
- Added support for reading the client's IP address from the `X-Forwarded-For` header when the source address is in the `controller.trusted_proxies` config option.
- Added field showing the IP address a host is connecting from.
- Added per player statistic tracking when they were first seen joining a particular server.
- Added ability to remotely restart and/or stop the controller and hosts.

### Changes

> - Fixed filenames with extentions that are invalid on Windows such as "con.zip" not being trated as invalid.

Reviewer: ```suggestion
- Fixed filenames with extensions that are invalid on Windows such as "con.zip" not being treated as invalid.
```

- Fixed generate host token modal not using a random id when the optional id field is unset.
- Exported Factorio data is now tied to mod packs instead of being per cluster.
- Fixed export failing if a mod uses a different root folder mod_version in its zip file.
- Fixed sorting of user last seen column in the Web UI.
- The list users permission is now granted by default to players.
- Fixed incorrect handling of hosts and instance with an id of 0 causing them to not update properly.

> - Fixed unable to unassiged instances in the Web UI.

Reviewer: ```suggestion
- Fixed unable to unassign instances in the Web UI.
```

- Fixed host crashing if the Factorio server is not marked executable.
- Fixed errors logged from instance not being attributed to the instance in certain situations.
- Hosts no longer print messages from instances into their console.
- Clusterio now treat unhandled Node.js errors and promise rejections as fatal and exit immediately.
- Fixed a potential for data loss after a hard system crash due to fsync not being called on files written.
- Fixed Factorio server crashing in the research_sync plugin when a mod enables a previously disabled technology and a player attempts to research it.
- Fixed Factorio server crashing when the game starts with tick_paused = true
- Removed migration of alpha 10's single file log format.
- Log errors linking mods on Windows instead of preventing startup.
- Added locale support for modules patched into to the game save.
- Added support for defining a module_exports.lua file for modules patched into the game save.

> - Fixed controller logs being split on server's local time day boundnary instead of UTC day boundnary.

Reviewer: ```suggestion
- Fixed controller logs being split on server's local time day boundary instead of UTC day boundary.
```

- Fixed Unicode BOM in mod locale files breaking instance data export.
- Added a warning to the server-setting.json that is automatically written to instance folders stating changes will be overwritten.

> - Changed save patching to use a `.tmp.zip` extention for the temporary file created instead of `zip.tmp`.

Reviewer: ```suggestion
- Changed save patching to use a `.tmp.zip` extension for the temporary file created instead of `zip.tmp`.
```
